src/main/java/org/asamk/signal/http/HttpServerHandler.java

```diff
@@ -0,0 +1,209 @@
+package org.asamk.signal.http;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ContainerNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.POJONode;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpServer;
+
+import org.asamk.signal.commands.Commands;
+import org.asamk.signal.commands.JsonRpcNamespace;
+import org.asamk.signal.commands.LocalCommand;
+import org.asamk.signal.commands.MultiLocalCommand;
+import org.asamk.signal.commands.RegistrationCommand;
+import org.asamk.signal.commands.exceptions.CommandException;
+import org.asamk.signal.jsonrpc.JsonRpcException;
+import org.asamk.signal.jsonrpc.JsonRpcRequest;
+import org.asamk.signal.jsonrpc.JsonRpcResponse;
+import org.asamk.signal.manager.Manager;
+import org.asamk.signal.manager.MultiAccountManager;
+import org.asamk.signal.manager.RegistrationManager;
+import org.asamk.signal.output.JsonWriterImpl;
+import org.asamk.signal.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.InetSocketAddress;
+import java.nio.channels.OverlappingFileLockException;
+import java.util.Map;
+import java.util.concurrent.Executors;
+
+public class HttpServerHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(HttpServerHandler.class);
+
+    private final ObjectMapper objectMapper = Util.createJsonObjectMapper();
+
+    private final int port;
+
+    private final Manager m;
+
+    private final MultiAccountManager c;
+
+    public HttpServerHandler(final int port, final Manager m) {
+        this.port = port;
+        this.m = m;
+        this.c = null;
+    }
+
+    public HttpServerHandler(final int port, final MultiAccountManager c) {
+        this.port = port;
+        this.m = null;
+        this.c = c;
+    }
+
+    public void init() {
+
+        try {
+
+            logger.info("Starting server on port " + port);
+
+            final var server = HttpServer.create(new InetSocketAddress(port), 0);
+            server.setExecutor(Executors.newFixedThreadPool(10));
+
+            server.createContext("/api/v1", httpExchange -> {
```

Comment:
Maybe `/api/v1/rpc` to be a bit more specific what this endpoint does.

Reply:
Makes sense. I changed that.